### DISCUSSION TITLE
阿波罗配置客户端更新失败及可观测性

### DISCUSSION THREAD

**Discussion by yearsFung** [apolloconfig/apollo#4038] [Q&A]
1. 请问阿波罗配置框架，本身是否有可观测性方面的功能？譬如说想知道在配置更新后，哪些客户端收到了更新，有无未更新失败的的客户端统计，进而可以触发告警信息。还是说可观测性是建议集成普罗米修斯之类的框架来实现？

2. 看了下源码，当前避免客户端更新失败的fallback机制是默认5min拉取一次配置。那是说默认情况下5min，有可能某些客户端存在配置数据不一致的情况，系统需要容忍这样的“问题”。除了减小间隔时间，增加拉取频率外，有无别的措施减少误差，告警什么的？好像代码实现上没有看到。

请大佬们明示，感谢~

**Comment by nobodyiam** [comment]
There is an instance list in the portal page:
![image](https://user-images.githubusercontent.com/837658/137568210-7bc1f4c1-a556-4f32-b2b1-096b83c95ff5.png)

There are also several tracer events and logs in the client-side, e.g. `syncRemoteConfig`, `Apollo.ConfigChangeListener` which you may find in `RemoteConfigRepository.java` and `AbstractConfig.java`. And you may refer [tracing](https://www.apolloconfig.com/#/zh/design/apollo-design?id=_51-tracing) for more information about how to integrate with some open source monitoring systems.

**Reply by yearsFung** [reply]
thanks!  I‘ll search and study it first
